src/Lecture2.hs

```diff
@@ -81,7 +94,10 @@ lists of even lengths.
 ♫ NOTE: Use eta-reduction and function composition (the dot (.) operator)
   in this function.
 -}
-evenLists = error "TODO"
+evenLists :: [[a]] -> [[a]]
+evenLists [] = []
+evenLists [x] = if even $ length x then [x] else []
+evenLists (x : xs) = evenLists [x] ++ evenLists xs
```

Comment:
This looks correct 👌🏻 
But you can implement this function even more elegantly using the standard function `filter` and the composition of functions `even` and `length` 👍🏻